pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -502,7 +492,7 @@ func (f *fakeChainReader) BatchGetLatestValues(_ context.Context, request types.
 	return result, nil
 }
 
-func (f *fakeChainReader) QueryKey(_ context.Context, _ string, filter query.KeyFilter, limitAndSort query.LimitAndSort, _ any) ([]types.Sequence, error) {
+func (f *fakeChainReader) QueryKey(_ context.Context, contractName string, filter query.KeyFilter, limitAndSort query.LimitAndSort, _ any) ([]types.Sequence, error) {
```

Comment:
out of scope for now -- this API needs pagination if the resulting sequence slice can be large

Reply:
It can be paginated through limit and sort since it returns a sequence cursor and allows limits